examples/testdata/wasi_arg.wat

```diff
@@ -0,0 +1,61 @@
+;; $wasi_arg is a WASI command which copies null-terminated args to stdout.
+(module $wasi_arg
+	;; args_get reads command-line argument data.
+	;;
+	;; See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-args_getargv-pointerpointeru8-argv_buf-pointeru8---errno
+    (import "wasi_snapshot_preview1" "args_get"
+        (func $wasi.args_get (param $argv i32) (param $argv_buf i32) (result (;errno;) i32)))
+
+	;; args_sizes_get returns command-line argument data sizes.
+	;;
+	;; See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-args_sizes_get---errno-size-size
+    (import "wasi_snapshot_preview1" "args_sizes_get"
+        (func $wasi.args_sizes_get (param $result.argc i32) (param $result.argv_buf_size i32) (result (;errno;) i32)))
+
+    ;; fd_write write bytes to a file descriptor.
+    ;;
+    ;; See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#fd_write
+    (import "wasi_snapshot_preview1" "fd_write"
+        (func $wasi.fd_write (param $fd i32) (param $iovs i32) (param $iovs_len i32) (param $result.size i32) (result (;errno;) i32)))
+
+    ;; WASI commands are required to export "memory". Particularly, imported functions mutate this.
+    ;;
+    ;; Note: 1 is the size in pages (64KB), not bytes!
+    ;; See https://www.w3.org/TR/2019/REC-wasm-core-1-20191205/#memories%E2%91%A7
+    (memory (export "memory") 1)
+
+    ;; _start is a special function defined by a WASI Command that runs like a main function would.
+    ;;
+    ;; See https://github.com/WebAssembly/WASI/blob/snapshot-01/design/application-abi.md#current-unstable-abi
+    (func $main (export "_start") (local $iovs i32) (local $ignored i32)
+        ;; $iovs are offset/length pairs in memory fd_write copies to the file descriptor.
+        ;; This function will only write one offset/length pair, corresponding to null-terminated args.
+        (local.set $iovs (i32.const 1024)) ;; 1024 is an arbitrary offset larger than the args.
```

Comment:
probably less clutter to define these as globals since you can do a constant initializer..